### PR TITLE
Add a demo for loading via NumPy

### PR DESCRIPTION
**Pull request by huonw**
This PR adds a demo that shows how to load node features from a NumPy array and an `IndexedArray`.

See: #1559

### CODE REVIEW

File: demos/basics/loading-numpy.ipynb [hunk 1]
```diff
@@ -0,0 +1,923 @@
+{
```
Comment (kjun9) [inline demos/basics/loading-numpy.ipynb:1]: <blockquote>Some algorithms work with than just a feature vector associated with each node.</blockquote><p>"Some algorithms work with more than just a feature vector associated with each node."</p><p> </p><p>Maybe also worth mentioning here that the dimensions in the multidimensional features are <code>N x F x T</code>? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/1599/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Comment (timpitman) [inline demos/basics/loading-numpy.ipynb:1]: <p>"This notebook walks through loading two kinds of graphs" lists 3 bullet points below?</p><p> </p><p>I think it should be "This notebook walks through loading three kinds of graphs:"</p><p> </p><p>Also there's two different paragraphs talking about NumPy vs Pandas - could they be combined?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/stellargraph/stellargraph/pull/1599/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

**Comment by huonw** [comment]
Do you have suggestions for what you mean by more obvious?

File: demos/basics/loading-numpy.ipynb (hunk 1 above)
Comment (huonw) [reply demos/basics/loading-numpy.ipynb:1]: I'm not sure what F and T are there? In particular, if F = features/variates and T = timesteps, it should be N x T x F.

Comment (huonw) [reply demos/basics/loading-numpy.ipynb:1]: <p>Could you be a bit more specific about which paragraphs you're talking about? There's the first one, the "StellarGraph supports loading data from ..." one and the "This notebook only uses NumPy for the node features ...", and all of them touch on this in some form.</p>

Comment (kjun9) [reply demos/basics/loading-numpy.ipynb:1]: Yep F = features/variates and T = timesteps. It says:

> For our square graph, we might have two time series of length three associated with each node

and I'm seeing
```
feature_tensors = np.array(
    [
        [[1.0, 1.0, 0.9], [-0.2, 0.1, 0.1]],
        [[2.0, 1.9, 2.1], [0.3, 0.31, 0.32]],
        [[3.0, 10.0, 3.0], [0.0, 0.0, 0.0]],
        [[4.0, 0.0, 1.0], [-0.5, -1.0, -3.0]],
    ],
    dtype=np.float32,
)
```
Isn't that 4 x 2 x 3? Which means it's N x F x T?

Comment (huonw) [reply demos/basics/loading-numpy.ipynb:1]: Ah, yeah, I guess that's a typo.

**Comment by timpitman** [comment]
> Do you have suggestions for what you mean by more obvious?

Originally I was thinking maybe e a different title, but reading the re-worded intro I think it's good now.

**Pull request merged**